### PR TITLE
migrate pkg/scheduler/internal/cache/debugger/comparer.go to structured logs

### PR DESCRIPTION
**Pull request by chymy**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
xref:
https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by chymy** [comment]
Thanks
/assign @k82cn @chendave

File: pkg/scheduler/internal/cache/debugger/comparer.go [hunk 1]
```diff
@@ -57,11 +57,11 @@ func (c *CacheComparer) Compare() error {
 	pendingPods := c.PodQueue.PendingPods()
 
 	if missed, redundant := c.CompareNodes(nodes, dump.Nodes); len(missed)+len(redundant) != 0 {
-		klog.Warningf("cache mismatch: missed nodes: %s; redundant nodes: %s", missed, redundant)
+	    klog.InfoS("Cache mismatch", "missedNodes", missed, "redundantNodes", redundant)
```
Comment (gavinfish) [inline pkg/scheduler/internal/cache/debugger/comparer.go:60]: do not change format. BTW, not sure about `missedNodes` such columns can provide real help. Leave it other reviewers.

File: pkg/scheduler/internal/cache/debugger/comparer.go [hunk 2]
```diff
@@ -39,8 +39,8 @@ type CacheComparer struct {
 
 // Compare compares the nodes and pods of NodeLister with Cache.Snapshot.
 func (c *CacheComparer) Compare() error {
-	klog.V(3).Info("cache comparer started")
-	defer klog.V(3).Info("cache comparer finished")
+	klog.V(3).Info("Cache comparer started")
```
Comment (ahg-g) [inline pkg/scheduler/internal/cache/debugger/comparer.go:42]: please keep as is, other places as well.

File: pkg/scheduler/internal/cache/debugger/comparer.go [hunk 3]
```diff
@@ -57,11 +57,11 @@ func (c *CacheComparer) Compare() error {
 	pendingPods := c.PodQueue.PendingPods()
 
 	if missed, redundant := c.CompareNodes(nodes, dump.Nodes); len(missed)+len(redundant) != 0 {
-		klog.Warningf("cache mismatch: missed nodes: %s; redundant nodes: %s", missed, redundant)
+		klog.InfoS("Cache mismatch", "missedNodes", missed, "redundantNodes", redundant)
```
Comment (ahg-g) [inline pkg/scheduler/internal/cache/debugger/comparer.go:60]: ditto, keep it as "missed nodes" and "redundant nodes", below too.

File: pkg/scheduler/internal/cache/debugger/comparer.go [hunk 4]
```diff
@@ -57,11 +57,11 @@ func (c *CacheComparer) Compare() error {
...
 	}
 
 	if missed, redundant := c.ComparePods(pods, pendingPods, dump.Nodes); len(missed)+len(redundant) != 0 {
-		klog.Warningf("cache mismatch: missed pods: %s; redundant pods: %s", missed, redundant)
+		klog.InfoS("Cache mismatch", "missedNodes", missed, "redundantNodes", redundant)
```
Comment (chendave) [inline pkg/scheduler/internal/cache/debugger/comparer.go:64]: this is not for nodes, it's pods, please revert it back.

File: pkg/scheduler/internal/cache/debugger/comparer.go (hunk 3 above)
Comment (chendave) [inline pkg/scheduler/internal/cache/debugger/comparer.go:60]: I cannot get the point of change "missed nodes" to "missedNodes", what's the consideration for this?

Comment (chendave) [reply pkg/scheduler/internal/cache/debugger/comparer.go:60]: +1

Comment (chymy) [reply pkg/scheduler/internal/cache/debugger/comparer.go:60]: if K has spaces, will it be confused?
eg:
```
"Cache mismatch" missed nodes=[node1 test1] redundant nodes=[node2 test2]
```
I will keep it as "missed nodes" and "redundant nodes". Thanks @ahg-g @chendave

File: pkg/scheduler/internal/cache/debugger/comparer.go [hunk 5]
```diff
@@ -57,11 +57,11 @@ func (c *CacheComparer) Compare() error {
 	pendingPods := c.PodQueue.PendingPods()
 
 	if missed, redundant := c.CompareNodes(nodes, dump.Nodes); len(missed)+len(redundant) != 0 {
-		klog.Warningf("cache mismatch: missed nodes: %s; redundant nodes: %s", missed, redundant)
+		klog.InfoS("cache mismatch", "missed nodes", missed, "redundant nodes", redundant)
 	}
 
 	if missed, redundant := c.ComparePods(pods, pendingPods, dump.Nodes); len(missed)+len(redundant) != 0 {
-		klog.Warningf("cache mismatch: missed pods: %s; redundant pods: %s", missed, redundant)
+		klog.InfoS("cache mismatch", "missed nodes", missed, "redundant nodes", redundant)
```
Comment (chendave) [inline pkg/scheduler/internal/cache/debugger/comparer.go:64]: missed pods, redundant pods

Comment (chymy) [reply pkg/scheduler/internal/cache/debugger/comparer.go]: Sorry.
Thank you very much @chendave . fixed.

**Review by chendave — COMMENTED** [review]
/lgtm

**Pull request merged**